### PR TITLE
Feature/round corner tooltip line

### PR DESCRIPTION
**Pull request by Uraskin**
Added the option to make round corners for data line.

### CODE REVIEW

**Comment by imaNNeoFighT** [comment]
Also, it doesn't work correctly,
Check the bottom of the line:

image

File: lib/src/chart/line_chart/line_chart_data.dart [hunk 1]
```diff
@@ -1347,6 +1347,9 @@ class LineTouchData extends FlTouchData with EquatableMixin {
   /// and goes through the targeted spot.
   final bool fullHeightTouchLine;
 
+  /// Sets the indicator line will have round corners on the ends, default 'false'.
+  final bool roundTouchLineConers;
```
[the remark this answers is not in the document]
Comment (Uraskin) [reply lib/src/chart/line_chart/line_chart_data.dart]: Fix that.

**Comment by Uraskin** [comment]
> Also, it doesn't work correctly,
> Check the bottom of the line:
> 
> image

Fix that
fix round cap

File: test/chart/data_pool.dart [hunk 2]
```diff
@@ -60,7 +60,7 @@ final SideTitles sideTitles1 = SideTitles(
 final SideTitles sideTitles1Clone = SideTitles(
   margin: 1,
   reservedSize: 10,
-  textStyle: TextStyle(color: Colors.green),
+  textStyle: const TextStyle(color: Colors.green),
```
Comment (imaNNeoFighT) [inline test/chart/data_pool.dart:63]: Please revert these changes.
They must be changed in a separate PR.

Comment (imaNNeoFighT) [reply test/chart/data_pool.dart:63]: Then please resolve the conflicts.
Thanks!

**Comment by DeD1rk** [comment]
> > Also, it doesn't work correctly,
> > Check the bottom of the line:
> >
> 
> Fix that 

Actually, I'd say it is expected behaviour that the rounded part of the line extends below the chart body. That would be similar to the behaviour of the data line itself: 
image

**Comment by imaNNeoFighT** [comment]
Is there any update here?
